api/src/pcapi/core/bookings/models.py

```diff
@@ -135,6 +135,7 @@ class Booking(PcObject, Base, Model):
             values_callable=lambda x: [reason.value for reason in BookingCancellationReasons],
         ),
         nullable=True,
+        index=True,
```

Comment:
On a actuellement en production plus de 40M de lignes dans la table `booking`, mais la majorité sans annulation : moins de 10M avec une raison d'annulation (requête : `select count(id) from booking where "cancellationReason" is not null`).

Ne serait-il donc pas judicieux de se limiter à un index partiel ?
Cela reviendrait à une déclaration d'`Index` séparée, et donc sur la recommandation de @dbaty qui peut aussi donner son avis ici :slightly_smiling_face: 

```
Index(
    "ix_booking_cancellation_reason",
    cancellationReason,
    postgresql_where=cancellationReason.is_not(None)
)
```

Le temps de création de l'index est-il raisonnable ? (testé sur staging ?)

Reply:
J'ai hésité à rajouter un index partiel. J'ai lancé la création de l'index sur staging ça a pris environ 3 minutes, du coup je l'ai laissé tel quel. Mais je peux quand même rajouter le filtre sur les valeurs non-null

Reply:
Ça me semble être une bonne idée, car l'index prendra moins de place sur le disque.